server/src/main/java/org/elasticsearch/script/field/IPAddress.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.script.field;
+
+import org.elasticsearch.common.network.InetAddresses;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.net.Inet4Address;
+import java.net.Inet6Address;
+import java.net.InetAddress;
+
+/**
+ * IP address for use in scripting.
+ */
+public class IPAddress implements ToXContent {
+    protected final InetAddress address;
+
+    IPAddress(InetAddress address) {
+        this.address = address;
+    }
+
+    public IPAddress(String address) {
+        this.address = InetAddresses.forString(address);
+    }
+
+    public boolean isV4() {
+        return address instanceof Inet4Address;
```

Comment:
Big thumbs up for this utility method along with the v6.  Annoying that the Java API is like this. (Nothing actionable, just like this solution.)

Reply:
Thanks, I had to provide these.

Without these two methods there would be no way for users to determine the version of address.